src/Sql/Sql/Elastic Jobs/Cmdlet/Target/AzureSqlElasticJobTargetCmdletBase.cs

```diff
@@ -0,0 +1,502 @@
+
+        /// <summary>
+        /// Intializes the model adapter
+        /// </summary>
+        /// <returns>The Azure Elastic Job Adapter</returns>
+        protected override AzureSqlElasticJobAdapter InitModelAdapter()
+        {
+            return new AzureSqlElasticJobAdapter(DefaultContext);
+        }
+
+        /// <summary>
+        /// Gets or sets the resource group name.
+        /// </summary>
+        [Parameter(
+            Mandatory = true,
+            Position = 0,
+            ParameterSetName = DefaultSqlServerOrElasticPoolSet,
+            HelpMessage = "The resource group name")]
+        [Parameter(
+            Mandatory = true,
+            Position = 0,
+            ParameterSetName = DefaultSqlDatabaseSet,
+            HelpMessage = "The resource group name")]
+        [Parameter(
+            Mandatory = true,
+            Position = 0,
+            ParameterSetName = DefaultSqlShardMapSet,
+            HelpMessage = "The resource group name")]
+        [ValidateNotNullOrEmpty]
+        [ResourceGroupCompleter]
+        public override string ResourceGroupName { get; set; }
+
+        /// <summary>
+        /// Gets or sets the target group input object.
+        /// </summary>
+        [Parameter(ParameterSetName = TargetGroupObjectSqlDatabaseSet,
+            Mandatory = true,
+            ValueFromPipeline = true,
+            Position = 0,
+            HelpMessage = "The target group object.")]
+        [Parameter(ParameterSetName = TargetGroupObjectSqlServerOrElasticPoolSet,
+            Mandatory = true,
+            ValueFromPipeline = true,
+            Position = 0,
+            HelpMessage = "The target group object.")]
+        [Parameter(ParameterSetName = TargetGroupObjectSqlShardMapSet,
+            Mandatory = true,
+            ValueFromPipeline = true,
+            Position = 0,
+            HelpMessage = "The target group object.")]
+        [ValidateNotNullOrEmpty]
+        public AzureSqlElasticJobTargetGroupModel ParentObject { get; set; }
+
+        /// <summary>
+        /// Gets or sets the target group resource id.
+        /// </summary>
+        [Parameter(ParameterSetName = ParentResourceIdSqlDatabaseSet,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 0,
+            HelpMessage = "The target group resource id.")]
+        [Parameter(ParameterSetName = ParentResourceIdSqlServerOrElasticPoolSet,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 0,
+            HelpMessage = "The target group resource id.")]
+        [Parameter(ParameterSetName = ParentResourceIdSqlShardMapSet,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 0,
+            HelpMessage = "The target group resource id.")]
+        [ValidateNotNullOrEmpty]
+        public string ParentResourceId { get; set; }
```

Comment:
@johnpaulkee please check if the `-ResourceId` case applies here in the base

Reply:
As discussed we'll address the Resource Id changes in a separate PR to reduce complexity of this one. We can do a one-off release with these changes and then add -ResourceId parameter  later.